Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -180,19 +187,24 @@ def _drawVarComposite(self, glyph, pen, isPointPen):
 
         for comp in glyph.components:
 
-            t = comp.transform.toTransform()
-            if isPointPen:
-                tPen = TransformPointPen(pen, t)
-            else:
-                tPen = TransformPen(pen, t)
-
             with self.glyphSet.pushLocation(
                 comp.location, comp.flags & VarComponentFlags.RESET_UNSPECIFIED_AXES
             ):
-                if isPointPen:
-                    self.glyphSet[comp.glyphName].drawPoints(tPen)
-                else:
-                    self.glyphSet[comp.glyphName].draw(tPen)
+                try:
+                    pen.addVarComponent(
+                        comp.glyphName, comp.transform, self.glyphSet.rawLocation
+                    )
+                except NotImplementedError:
+                    t = comp.transform.toTransform()
+                    if isPointPen:
+                        tPen = TransformPointPen(pen, t)
+                    else:
+                        tPen = TransformPen(pen, t)
+
+                        if isPointPen:
```

Comment:
Found the problem: this section is indented one level too deep, so it only works for a segment pen.